src/widgets/rss/RssWidgetTile.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  ActionIcon,
+  Badge,
+  Card,
+  Center,
+  createStyles,
+  Flex,
+  Group,
+  Image,
+  Loader,
+  LoadingOverlay,
+  MediaQuery,
+  ScrollArea,
+  Stack,
+  Text,
+  Title,
+  UnstyledButton,
+} from '@mantine/core';
+import { useElementSize } from '@mantine/hooks';
+import {
+  IconBulldozer,
+  IconCalendarTime,
+  IconClock,
+  IconCopyright,
+  IconRefresh,
+  IconRss,
+  IconSpeakerphone,
+} from '@tabler/icons';
+import { useTranslation } from 'next-i18next';
+import Link from 'next/link';
+import { useState } from 'react';
+import { useGetRssFeed } from '../../hooks/widgets/rss/useGetRssFeed';
+import { sleep } from '../../tools/client/time';
+import { defineWidget } from '../helper';
+import { IWidget } from '../widgets';
+
+const definition = defineWidget({
+  id: 'rss',
+  icon: IconRss,
+  options: {
+    rssFeedUrl: {
+      type: 'text',
+      defaultValue: '',
+    },
+  },
+  gridstack: {
+    minWidth: 2,
+    minHeight: 2,
+    maxWidth: 12,
+    maxHeight: 12,
+  },
+  component: RssTile,
+});
+
+export type IRssWidget = IWidget<(typeof definition)['id'], typeof definition>;
+
+interface RssTileProps {
+  widget: IRssWidget;
+}
+
+function RssTile({ widget }: RssTileProps) {
+  const { t } = useTranslation('modules/rss');
+  const { data, isLoading, isFetching, refetch } = useGetRssFeed();
+  const { classes } = useStyles();
+  const [loadingOverlayVisible, setLoadingOverlayVisible] = useState(false);
+  const { ref, height } = useElementSize();
+
+  if (!data || isLoading) {
+    return (
+      <Center>
+        <Loader />
+      </Center>
+    );
+  }
+
+  if (!data.success) {
+    return (
+      <Center style={{ height: '100%' }}>
+        <Stack align="center">
+          <IconRss size={40} strokeWidth={1} />
+          <Title order={6}>{t('card.errors.general.title')}</Title>
+          <Text align="center">{t('card.errors.general.text')}</Text>
+        </Stack>
+      </Center>
+    );
+  }
+
+  return (
+    <Stack ref={ref} style={{ height: '100%' }}>
+      <LoadingOverlay visible={loadingOverlayVisible} />
+      <Flex gap="md">
+        {data.feed.image ? (
+          <Image
+            src={data.feed.image.url}
+            alt={data.feed.image.title}
+            width="auto"
+            height={40}
+            maw="50%"
+            mx="auto"
+          />
+        ) : (
+          <Title order={6}>{data.feed.title}</Title>
+        )}
+        <UnstyledButton
+          onClick={async () => {
+            setLoadingOverlayVisible(true);
+            await sleep(1500);
```

Comment:
Would make more sense to use a Promise.all for those two awaits. Then it will take at least 1.5 seconds to load, but not additional 1.5 seconds before loading.

Reply:
Good catch!